tests/1d_flow.cc

```diff
@@ -0,0 +1,238 @@
+// distribution.
+//
+// --------------------------------------------------------------------------
+
+#include <deal.II/base/function.h>
+#include <deal.II/base/function_time.h>
+#include <deal.II/base/index_set.h>
+#include <deal.II/base/logstream.h>
+#include <deal.II/base/parameter_handler.h>
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/base/utilities.h>
+
+#include <deal.II/dofs/dof_tools.h>
+
+#include <deal.II/fe/fe_q.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_in.h>
+#include <deal.II/grid/grid_out.h>
+#include <deal.II/grid/manifold_lib.h>
+#include <deal.II/grid/tria.h>
+#include <deal.II/grid/tria_accessor.h>
+#include <deal.II/grid/tria_iterator.h>
+
+#include <deal.II/lac/affine_constraints.h>
+#include <deal.II/lac/la_parallel_vector.h>
+
+#include <deal.II/numerics/data_out.h>
+#include <deal.II/numerics/vector_tools.h>
+
+#include <adaflo/navier_stokes.h>
+#include <adaflo/time_stepping.h>
+
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+
+using namespace dealii;
+
+/**
+ * Simple test case for 1d Navier-Stokes
+ *
+ *
+ * velocity(t=0) = 2
+ *                             L=2.5
+ * (pressure=2) +-------------------------------+ (p=1)
+ *              |--> x
+ *
+ */
+
+
+template <int dim>
+class InflowVelocity : public Function<dim>
+{
+public:
+  InflowVelocity()
+    : Function<dim>()
+  {}
+
+  double
+  value(const Point<dim> &p, const unsigned int component = 0) const override
+  {
+    (void)p;
+    (void)component;
+    if constexpr (dim == 1)
+      return 2.0;
+    else
+      AssertThrow(false, ExcMessage("Advection field for dim!=1 not implemented"));
+  }
+};
+
+
+template <int dim>
+class ChannelFlow
+{
+public:
+  ChannelFlow(const FlowParameters &parameters);
+  void
+  run();
+
+private:
+  void
+  output_results() const;
+
+  ConditionalOStream pcout;
+
+  mutable TimerOutput timer;
+
+  Triangulation<dim> triangulation;
+  NavierStokes<dim>  navier_stokes;
+};
+
+template <int dim>
+ChannelFlow<dim>::ChannelFlow(const FlowParameters &parameters)
+  : pcout(std::cout, (Utilities::MPI::this_mpi_process(MPI_COMM_WORLD) == 0))
+  , timer(pcout, TimerOutput::summary, TimerOutput::cpu_and_wall_times)
+  , navier_stokes(parameters, triangulation, &timer)
+{}
+
+template <int dim>
+void
+ChannelFlow<dim>::output_results() const
+{
+  timer.enter_subsection("Generate output.");
+
+  navier_stokes.output_solution(navier_stokes.get_parameters().output_filename);
+
+  timer.leave_subsection();
+}
+
+
+template <int dim>
+void
+create_triangulation(Triangulation<dim> &tria)
+{
+  AssertThrow(dim == 1, ExcNotImplemented());
+  GridGenerator::hyper_rectangle(tria, Point<dim>(0.0), Point<dim>(2.5));
+  tria.refine_global(10);
+
+  for (const auto &cell : tria.cell_iterators())
+    for (auto &face : cell->face_iterators())
+      if ((face->at_boundary()))
+        {
+          if (face->center()[0] == 0)
```

Comment:
This comparison seems a bit dangerous due to floating point nature. I would prefer
```suggestion
          if (std::abs(face->center()[0]) < 1e-12)
```

Reply:
done.